geodiff/src/geodiff.cpp

```diff
@@ -693,6 +693,111 @@ int GEODIFF_dumpData( const char *driverName, const char *driverExtraInfo, const
 }
 
 
+int GEODIFF_schema( const char *driverName, const char *driverExtraInfo, const char *src, const char *json )
+{
+  std::unique_ptr<Driver> driver( Driver::createDriver( std::string( driverName ) ) );
+  if ( !driver )
+  {
+    Logger::instance().error( "Cannot create driver " + std::string( driverName ) );
+    return GEODIFF_ERROR;
+  }
+
+  try
+  {
+    // open source
+    std::map<std::string, std::string> conn;
+    conn["base"] = std::string( src );
+    if ( driverExtraInfo )
+      conn["conninfo"] = std::string( driverExtraInfo );
+    driver->open( conn );
+
+    // prepare JSON
+    std::vector<std::string> tablesData;
+    for ( const std::string &tableName : driver->listTables() )
+    {
+      TableSchema tbl = driver->tableSchema( tableName );
+
+      std::vector<std::string> columnsJson;
+      for ( const TableColumnInfo &column : tbl.columns )
+      {
+        std::vector<std::string> columnData;
+        columnData.push_back( "\"name\": \"" + jsonQuoted( column.name ) + "\"" );
+        columnData.push_back( "\"type\": \"" + TableColumnType::baseTypeToString( column.type.baseType ) + "\"" );
+        columnData.push_back( "\"type_db\": \"" + jsonQuoted( column.type.dbType ) + "\"" );
+        if ( column.isPrimaryKey )
+          columnData.push_back( "\"primary_key\": true" );
```

Comment:
dont you want to add to output also "false" option, so the json has always the same keys?